src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -730,3 +730,78 @@ def _find_property(instance, path):
     for part in path:
         instance = _update_instance(instance, part, path)
     return instance
+
+
+def assign_implict_identity(getter, setter, identity_role=None, identity_scope=None):
+    import time
+    from azure.mgmt.authorization import AuthorizationManagementClient
+    from azure.mgmt.authorization.models import RoleAssignmentProperties
+    from msrestazure.azure_exceptions import CloudError
+
+    # get
+    resource = getter()
+    if resource.identity:
+        logger.warning('Implict identity is already configured')
+    else:
+        resource = setter(resource)
+
+    # create role assignment:
+    if identity_scope:
+        principal_id = resource.identity.principal_id
+
+        identity_role_id = resolve_role_id(identity_role, identity_scope)
+        assignments_client = get_mgmt_service_client(AuthorizationManagementClient).role_assignments
+        properties = RoleAssignmentProperties(identity_role_id, principal_id)
+
+        logger.info("Creating an assignment with a role '%s' on the scope of '%s'", identity_role_id, identity_scope)
+        retry_times = 36
```

Comment:
Retry 36 times! 1) Magic number? Any guidelines coming from identity team? 2) 36 times !?

Reply:
So it's 3 minutes ...

Reply:
Yep, recommended by the PAS team to workaround a known server replication delay :)